File: day1.lua
sums = {}
sum = 0
for line in io.lines('day1.txt') do
	if line ~= '' then
		sum = sum + tonumber(line)
	else
		table.insert(sums,sum)
		sum = 0
	end
end
print(sum)
table.insert(sums,sum)	
table.sort(sums)

-- day 1 part 1
print(sums[#sums])

topThree = 0
for i = 0,2 do
	topThree = topThree + sums[#sums-i]
end

-- day 1 part 2
print(topThree)

File: day1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int comparator(const void* a, const void* b){return *(int*)a<*(int*)b;}

int main(void) {
  int sum = 0, topThree = 0, top[1000],count=0;
  char nums[100];
  FILE *input = fopen("./day1.txt", "r");
  do {
    fgets(nums, 10, input);
    if (strcmp(nums, "\n") == 0) {
      top[count] = sum;
      count++;
      sum = 0;
    } else {
      sum += atoi(nums);
    }
  } while (!feof(input));
  top[count] = sum;
  
  qsort(top, count, sizeof(int), comparator);
  
  printf("Part 1: %d", top[0]);
  
  for(int i = 0;i<3;i++){topThree+=top[i];}
  printf("Part 2: %d", topThree);
  
  fclose(input);
  return 0;
}
